### ISSUE TITLE
mdBook include directive vs `tutorial_generator`

### ISSUE THREAD

**Issue by simonsan** [LukeMathWalker/pavex#152]
I read the newsletter https://www.lpalmieri.com/posts/this-month-in-pavex-08/ and saw https://github.com/LukeMathWalker/pavex/blob/main/doc_examples/quickstart/tutorial.yml , and I wondered if mdBook has been considered over mkdocs?

mdBook can include portions of a file: https://rust-lang.github.io/mdBook/format/mdbook.html?#including-portions-of-a-file which means, that code can live separate from the actual documentation and can be tested as such. Also, `mdbook test` can test rust code examples embedded in Markdown.

So what I wondered reading the blog post, has mdBook been considered and if so, what was the reasoning to decide against it? I'm interested in it, solely for my own research purposes.

**Comment by LukeMathWalker** [comment]
It has been considered—I have used it extensively in the past (e.g. "Zero to Production in Rust" is an mdbook).

It's great for book-like content, but I don't feel like it's a good fit for documentation websites. The UI, in particular, is not what I was looking for.  
Material for mkdocs ticked all the boxes from the get-go so I went straight for it.

On testing: it's significantly more complex that just testing the snippet that's shown in the docs. The snippet is just a small part of the actual code and, in the case of Pavex, we don't want to run `cargo test`.
